### DISCUSSION TITLE
Separate options for Video and Storyboard

### DISCUSSION THREAD

**Discussion by ryancsh** [ppy/osu#13134] [Ideas] [answered]
**_Separate options for Video and Storyboard_**

The short term reason is so that players can avoid the massive fps drops from 400+ to 60.

The longer term reason is somebody might just not like storyboards.
Some storyboards have a lot of flashing, which can be a potential health concern for somebody prone to seizures whereas videos don't tend to have a lot of flashing from what I can tell.

**Comment by peppy** [comment]
Storyboards will be optimised to the point there is no frame drop.

I'm not sure "flashing" is a concern. There nothing saying a video won't have flashing.

**Comment by bdach** [comment]
Also see #3114 for the flashing part.

**Comment by ryancsh** [comment]
> 
> I'm not sure "flashing" is a concern. There nothing saying a video won't have flashing.

There is no guarantee obviously, but I find leaving video enabled doesn't distract from the gameplay while giving me something to watch when I'm waiting for breaks in the middle of a map to end.

Storyboards on the other hand tend to actually distract from the game itself. And some people go way overboard with the effects. 

My perspective is like so: Having separate options for them doesn't hurt but can potentially help. Assuming the effort to make it happen isn't too much, why not?

**Comment by peppy** [comment, marked as answer]
Simplification; unification of experience; mapper intent.

**Comment by Scraft161** [comment]
this is my experience from using stable (in wine on linux)
storyboards don't add noticeable lag (on top of what wine gave) and provide nice features if done right.
video's on the other hand tend to be extremely laggy on both stable and lazer due to the overhead of video decoding.
I love using storyboards as to not have to look at the same backdrop the whole time and if done right can add to the user experience, add to this that most maps that include video often ship with a storyboard for those who prefer (again from stable which DID have separate toggles for this)

if the issue is that the options screen would become too complicated with small things like this then allow these to be set in the options file and add a note in the settings menu for that.
lot's of people like tweaking their game to make it fit their way of enjoying it to the max (looking at you fellow linux users).

**Reply by peppy** [reply]
That sounds back to front. And a wine specific issue. Please actually test outside of emulation layers before commenting

**Reply by Scraft161** [reply]
I haven't been able to test outside of wine as I don't own a computer running windows, but I did find the cause for it on stable.
normally video decoding should be pushed to the GPU whereas somewhere this isn't happening through wine, the GPU is detected and accessible but not used, whether this is to blame on osu, the video decoding library, wine, or NVIDIA is unclear.
what I can conclude is that video decoding can be very taxing when not offloaded.

this brings me to my next point, while I couldn't force stable to decode video in GPU, I was able to get lazer to decode on CPU only and found similar results on Linux natively.
I wish I could test the other way around too, but I don't have access to a windows PC nor am I able to set up a VM with GPU access.
I will try to see if I can get a bootable drive with GPU drivers and test it as soon as I can.

lastly I want to do a bit of speculation a to why storyboards tend to be laggy, but I could be far from correct.
what I think is that video decoding is being done by a library and can easily be pushed of to the GPU.
but storyboards on the other hand require instructions to be parsed by the CPU and either parsing those instructions or handing them over to openGL (or DirectX's equivalent) is done in a way that isn't made for how storyboards are handled.
this could result in slow handling on GPU as there is data being transferred between RAM and VRAM, but is still faster compared to video decoding on the CPU.
feel free to correct me on any of this if I'm wrong, I'm honestly quite curious as to how all of this is being handled and how this could be optimized.

**Reply by peppy** [reply]
We do have gpu video decoding in a branch somewhere, but at the resolution we're decoding the overhead should be minimal unless you're running on potato.

**Reply by Scraft161** [reply]
I think this is just library funkiness on Linux then, I'll definitely look into it.
Thanks for your time.
